refract-spec.md

```diff
@@ -12,89 +12,286 @@ This document conforms to RFC 2119, which says:
 
-## Data Structure
+## Primitive Types
 
-Refract provides a single structure for data, which will be referred to throughout this document as an element. All elements found in a Refract document SHOULD conform to the element structure as defined here.
+Primitive Types are types used by Refract as the basis for all values and structure. These types use both JSON primitive values along with Refract Elements.
 
-### Element
+Refract provides Primitive Types and Primitive Elements for the purpose of expanding JSON primitive values and adding meta data and attributes. The allows Refract to be infinitely recursive so meta data and attributes can always be added to any value.
 
-The Refract Element contains three properties, `element`, `attributes`, and `content`, as defined below. This Element can be used recursively throughout the document, even as a value for each of its own properties.
+### Value Types
 
-#### Properties
+Value types are primitive types that Refract uses that do not define structure. Value types consist of:
 
-- `element` (enum, required)
+* Null Type
+* String Type
+* Number Type
+* Boolean Type
 
-    The `element` property defines the type of element. It MAY be a string that references an element type, or it MAY be an `element` itself that provides additional data about the element type.
+#### Null Type (enum)
 
-    - Members
-        - (string)
+A Null Type in Refract can be either the `null` value or a Null Element.
+
+##### Members
+
+- (null)
+- (Null Element)
+
+#### String Type (enum)
+
+A String Type in Refract can be either a string value or a String Element.
+
+##### Members
+
+- (string)
+- (String Element)
+
+#### Number Type (enum)
+
+A Number Type in Refract can be either a number value or a Number Element.
+
+##### Members
+
+- (number)
+- (Number Element)
+
+#### Boolean Type (enum)
+
+A Boolean Type in Refract can be either a boolean value or a Boolean Element.
+
+##### Members
+
+- (boolean)
+- (Boolean Element)
+
+### Structure Types
+
+Structure Types provide structure for Refract documents. These types consist of:
+
+* Array Type
+* Object Type
+* Member Type
+
+#### Array Type (array)
+
+An Array Type MAY be an array of all Refract primitives except the Object Type. This ensures there is no confusion between an object and an actual Refract Object Type.
+
+##### Members
+
+- (enum)
+    - (Null Type)
+    - (String Type)
+    - (Number Type)
+    - (Boolean Type)
+    - (Array Type)
+    - (Element)
+
+##### Example
+
+This example shows an array containing a sample of each primitive type along with an element.
+
+```json
+[
+  null,
+  "foo",
+  100,
+  true,
+  [1, 2, 3],
+  {
+    "element": "string",
+    "content": "bar"
+  }
+]
+```
+
+#### Object Type (enum)
+
+Object Types provide a key/value structure for Refract.
+
+When converting an Object Type to a primitive JSON type, and the Object Type is a JSON object, no processing is needed. If the Object Type is an array, the following rules apply:
```

Comment:
Again, what if I am in ruby and working with hashes. Do we want to use this type of wording?